addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -300,6 +315,43 @@ export class LinkPlugin extends Plugin {
             link = this.getOrCreateLink();
         }
         this.linkElement = link;
+        this.handleImage();
+    }
+
+    handleImage() {
+        const selectionData = this.shared.getSelectionData();
+        if (selectionData?.editableSelection?.isCollapsed) {
```

Comment:
I don't think the optional chaining here is necessary since we are sure there is an object returning from getSelectionData, also the logic behind that wouldn't work since if selectionData is null or editableSelection is null we don't want to run the function too but in this case it will skip the return

Reply:
makes sense if thats how it works.